python/glow/gwas/lin_reg.py

```diff
@@ -186,14 +220,18 @@ def _linear_regression_inner(genotype_pdf: pd.DataFrame, Y_state: YState,
     So, if a matrix's indices are `sg` (like the X matrix), it has one row per sample and one column per genotype.
     '''
 
-    X = np.column_stack(genotype_pdf[_VALUES_COLUMN_NAME].array)
+    genotype_values = genotype_pdf[_VALUES_COLUMN_NAME].array
+    if gt_indices_to_drop is not None and gt_indices_to_drop.size:
+        genotype_values = list(map(lambda x: np.delete(x, gt_indices_to_drop), genotype_values))
+    X = np.column_stack(genotype_values)
```

Comment:
You can mask out the samples after this line like `np.delete(X, gt_indices_to_drop, axis=0)`. It's best to avoid pure Python loops whenever possible.